### PR TITLE
Allow trailing whitespace on sparse matrix (SCP-4063)

### PR DESCRIPTION
**Pull request by ehanna4**
Scanpy and ingest pipeline don't care if a sparse matrix has trailing whitespace, they just ignore it so this updates the CSFV to also ignore that.

To test:
- pull this branch
- attempt to upload a sparse matrix that has trailing white spaces on any of the rows and there should be no error

Also updated a test file that had the incorrect number of lines listed.

### CODE REVIEW

File: app/javascript/lib/validation/expression-matrices-validation.js [hunk 1]
```diff
@@ -353,13 +356,12 @@ function validateValuesAreNumeric(line, isLastLine, lineNum, dataObj) {
 }
 
 /**
-   * Validate the first line in the sparse Matrix begins with the string '%%MatrixMarket'
+   * Validate the first line in the sparse Matrix begins with '%%MatrixMarket'
```
Comment (eweitz) [inline app/javascript/lib/validation/expression-matrices-validation.js:359]:
```suggestion
   * Validate the first line in the sparse matrix begins with '%%MatrixMarket'
```

File: app/javascript/lib/validation/expression-matrices-validation.js [hunk 2]
```diff
@@ -353,13 +356,12 @@ function validateValuesAreNumeric(line, isLastLine, lineNum, dataObj) {
...
    */
 function validateMTXHeaderLine(line) {
   const issues = []
-  const mtxHeader = line.slice(0, 14)
-  if (mtxHeader !== '%%MatrixMarket') {
-    const msg = `First line must begin with "%%MatrixMarket", not "${mtxHeader}"`
+  if (line[0] !== '%%MatrixMarket') {
+    const msg = `First line must begin with "%%MatrixMarket", not "${line}"`
```
Comment (eweitz) [inline app/javascript/lib/validation/expression-matrices-validation.js:364]: Blocker: we'll want to print the first element in `line`, not the whole array.

```suggestion
    const msg = `First line must begin with "%%MatrixMarket", not "${line[0]}"`
```

**Review by eweitz — CHANGES_REQUESTED** [review]
Nice refinement!  I found a trivial blocker.

**Review by bistline — APPROVED** [review]

**Review by devonbush — APPROVED** [review]
If you can do it in <15 minutes, please add a js unit test for this case.  Otherwise, looks good.

**Review by eweitz — APPROVED** [review]

**Pull request merged**